app/assets/javascripts/discourse/app/components/post/meta-data/poster-name.gjs

```diff
@@ -25,15 +25,17 @@ export default class PostMetaDataPosterName extends Component {
   showNameAndGroup = true;
   showGlyph = true;
 
-  constructor() {
-    super(...arguments);
-    this.#trackUserStatus();
-  }
+  trackUserStatus = modifier((_, [user]) => {
```

Comment:
Modifiers are intended for modifying dom elements, so it's not great to use them for this kind of thing.

I think `helperFn` might be a better fit here? Example use [here](https://github.com/discourse/discourse/blob/c7f68715b59f955afb9a8a64786f0016b9aaf21f/app/assets/javascripts/discourse/app/components/user-tip.gjs#L18C4-L29C6). The implementation is essentially the same, but it runs in the 'render' phase of the runloop instead of after-render